### PR TITLE
[docs] review and authentication docs

### PR DESCRIPTION
**Pull request by jamiltz**

### CODE REVIEW

**Comment by jamiltz** [comment]
@pasin the second commit (https://github.com/couchbase/couchbase-lite-ios/pull/2138/commits/5e697f6d4788f44443605d030ffb77c6591b7e07) probably needs a review. The first commit is mostly presentational changes so doesn't need to be reviewed.

File: Swift/Tests/SampleCodeTest.swift [hunk 1]
```diff
@@ -496,6 +496,30 @@ class SampleCodeTest: CBLTestCase {
...
+        let target = URLEndpoint(url: url)
+        let config = ReplicatorConfiguration(database: database, target: target)
+        config.authenticator = BasicAuthenticator(username: "john", password: "pass")
+
+        self.replicator = Replicator(config: config)
+        self.replicator.start()
+        // # end::basic-authentication[]
+    }
+
+    func dontTestReplicationSessionAuthentication() throws {
```
Comment (pasin) [inline Swift/Tests/SampleCodeTest.swift]: We have SessionAuthenticator for this. Can you change to use that?

Comment (jamiltz) [reply Swift/Tests/SampleCodeTest.swift]: Ok - I'm a bit confused reading comments in https://issues.couchbase.com/browse/CBSE-5104

When should we recommended to use SessionAuthenticator vs. setting the cookie in the header?

Comment (pasin) [reply Swift/Tests/SampleCodeTest.swift]: For SG Session, we should recommend to use SessionAuthenticator. The customer header is for something in addition e.g. proxy required headers or cookies.

**Comment by jamiltz** [comment]
@pasin updated session auth example (https://github.com/couchbase/couchbase-lite-ios/pull/2138/commits/2d6c5942fa0e5f3afc4d0c9c724c8ade803237c2). I think this PR can now be merged. Thanks!

**Pull request merged**